kolena/_experimental/dataset/_evaluation.py

```diff
@@ -184,12 +188,79 @@ def validate_data(left: pd.DataFrame, right: pd.DataFrame) -> None:
         raise IncorrectUsageError("numbers of rows between two dataframe do not match")
 
 
+def _validate_on(left: pd.DataFrame, right: pd.DataFrame, on: TEST_ON_TYPE) -> None:
+    if on is None:
+        raise IncorrectUsageError("on cannot be None")
+    # works for both string and list of string
+    if len(on) == 0:
+        raise IncorrectUsageError("on cannot be empty")
+
+    if isinstance(on, str):
+        if on not in left.columns or on not in right.columns:
+            raise IncorrectUsageError(f"column {on} doesn't exist in target dataframe")
+    elif isinstance(on, list):
+        for col in on:
+            if col not in left.columns or col not in right.columns:
+                raise IncorrectUsageError(f"column {col} doesn't exist in target dataframe")
+
+
+def _get_default_infer_func(df_inf: pd.DataFrame, on: TEST_ON_TYPE) -> INFER_FUNC_TYPE:
+    if isinstance(on, str):
+        on = [on]
+
+    def infer(datapoints: pd.DataFrame) -> pd.DataFrame:
+        inferences = datapoints[on].merge(df_inf, how="left", on=on)
+        return inferences
+
+    return infer
+
+
+def _get_default_eval_func(df_metrics: pd.DataFrame, on: TEST_ON_TYPE) -> EVAL_FUNC_TYPE:
+    if isinstance(on, str):
+        on = [on]
+
+    def eval(
+        datapoints: pd.DataFrame,
+        inferences: pd.DataFrame,
+        eval_config: Optional[TYPE_EVALUATION_CONFIG],
+    ) -> pd.DataFrame:
+        metrics = datapoints[on].merge(df_metrics, how="left", on=on)
```

Comment:
This seems to imply `inferences` is not needed for metrics computation, is this right as a default?

Reply:
i think for the simplest use case, mapping metrics only with datapoints makes sense for me, we intentionally ignored the inferences and eval_config at here.
the [partial uploading support](https://linear.app/kolena/issue/KOL-3670/sdk-support-partial-uploading-inference-and-metrics) might be able to suppress potential errors

Reply:
How does that make sense? Metrics evaluation is doing some computation based on groundtruth and model inference results. Users have to read codes here to know that we are joining on datapoint by default. cc @y27choi for more thoughts.

Side question: this line here would return both datapoint columns and metrics conlumns, right?

Reply:
i think under the safe config, dp <=> inf <=> metrics is one to one mapping. 
this is the tricky part when exposing the `on`, if we want to accommodate gt or eval_config, we will need multiple `on` conceptually.

for the side note: it will only return metric columns due to `datapoints[on]`